osu.Game/Rulesets/Scoring/HitEventExtensions.cs

```diff
@@ -65,6 +70,16 @@ public static class HitEventExtensions
             return timeOffsets.Average();
         }
 
-        public static bool AffectsUnstableRate(HitEvent e) => !(e.HitObject.HitWindows is HitWindows.EmptyHitWindows) && e.Result.IsHit();
+        public static bool AffectsUnstableRate(HitEvent e) => AffectsUnstableRate(e.HitObject, e.Result);
+        public static bool AffectsUnstableRate(HitObject hitObject, HitResult result) => hitObject.HitWindows != HitWindows.Empty && result.IsHit();
+
+        public class UnstableRateCalculationResult
```

Comment:
I'm sure people are going to want changes to this, but I'm not sure how strict review will be so I've just left it without any safeties initially.

Reply:
Actually doesn't require off-by-one adjustments since it already kinda is the event count (at least post-loop-execution).